src/test/java/org/springframework/data/elasticsearch/core/MappingBuilderTests.java

```diff
@@ -409,6 +412,41 @@ public void shouldUseFieldNameOnMultiField() throws IOException, JSONException {
 		assertEquals(expected, mapping, false);
 	}
 
+	@Test // DATAES-639
+	public void shouldUseIgnoreAbove(){
+		// given
+		elasticsearchTemplate.createIndex(IgnoreAboveEntity.class);
+		elasticsearchTemplate.putMapping(IgnoreAboveEntity.class);
+
+		// when
+		Map mapping = elasticsearchTemplate.getMapping(IgnoreAboveEntity.class);
+		Map properties = (Map) mapping.get("properties");
+		Map message = (Map) properties.get("message");
+
+		// then
+		java.lang.reflect.Field field = ReflectionUtils.findField(IgnoreAboveEntity.class, "message");
+		assertThat(field).isNotNull();
+		Field fieldAnnotation = AnnotationUtils.getAnnotation(field, Field.class);
+		assertThat(fieldAnnotation).isNotNull();
+		assertThat(message.get("ignore_above")).isEqualTo(fieldAnnotation.ignoreAbove());
+	}
+
+	/**
+	 * @author Xiao Yu
+	 */
+	@Setter
+	@Getter
+	@NoArgsConstructor
+	@AllArgsConstructor
+	@Builder
+	@Document(indexName = "test-ignore-above-mapping-builder", type = "test", shards = 1, replicas = 0, refreshInterval = "-1")
```

Comment:
`shards = 1`is the default since Elasticsearch 7.0 and Spring Data Elasticsearch 4.0.x